timex/timex.go

```diff
@@ -0,0 +1,81 @@
+// Package timex provides stdlib time package related utilities.
+//
+// It exports type Ticker which works like stdlib time.Ticker but closes
+// its channel on stop.
+package timex
+
+import (
+	"context"
+	"time"
+)
+
+// A Ticker holds a channel that delivers 'ticks' of a clock at
+// intervals, similar to time.Ticker from the standard library.
+type Ticker struct {
+	C    <-chan time.Time
+	done chan struct{}
+}
+
+// NewTicker returns a new Ticker containing a channel that will send
+// the time with a period specified by the duration argument, similar to
+// standard library's time.Ticker
```

Comment:
Add a new paragraph: "NewTicker will panic if the duration is negative.".

Reply:
done.